src/Pages/ExpensesGraph.cshtml

```diff
@@ -0,0 +1,63 @@
+﻿@page
+@using System.Globalization
+@using ePiggyWeb.Utilities
+@model ePiggyWeb.Pages.ExpensesGraphModel
+
+
+<form method="get" class="mt-5">
+    <input type="hidden" name="handler" value="Filter" />
+    <input for="startDate" asp-for="StartDate" type="date" value="@Model.StartDate.ToShortDateString()" />
+    <input for="endDate" asp-for="EndDate" type="date" value="@Model.EndDate.ToShortDateString()" />
+    <input type="submit" value="Filter" class="btn btn-warning ml-5" />
+</form>
+
+
+<div class="row ml-2 mr-2 mt-3 mb-5">
+    <div class="col-sm" id="graphPie"><div class="spinner-border text-warning loader" role="status">
+        <span class="sr-only">Loading...</span>
+    </div></div>
+    <div class="col-sm" id="graphColumn"><div class="spinner-border text-danger loader" role="status">
+        <span class="sr-only">Loading...</span>
+    </div></div>
+</div>
+
+
+
+
+    <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+<script type="text/javascript">
+    // Load google charts
+    google.charts.load('current', { 'packages': ['corechart'] });
+    google.charts.setOnLoadCallback(drawChart);
+
+    // Draw the chart and set the chart values
+    function drawChart() {
+        var data = google.visualization.arrayToDataTable([
+            ['Importance', 'Amount'],
+            ['Necessary', @Model.Expenses.GetBy(Importance.Necessary).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['High', @Model.Expenses.GetBy(Importance.High).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Medium', @Model.Expenses.GetBy(Importance.Medium).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Low', @Model.Expenses.GetBy(Importance.Low).GetSum().ToString(CultureInfo.InvariantCulture)],
+            ['Unnecessary', @Model.Expenses.GetBy(Importance.Unnecessary).GetSum().ToString(CultureInfo.InvariantCulture)]
```

Comment:
Leaving a comment here even though I commented up above, we have a NumberFormatter utility function to display currency for every locale, if it's used here to display currency.

Reply:
Or you could you use this:
https://docs.microsoft.com/en-us/globalization/locale/number-formatting-in-dotnet-framework
Which I guess would make more sense, I'm not sure if the number formatting I've created before is actually useful :D

Reply:
Will take a look at it, thanks!

Reply:
Since this is for graph numbers and not only displaying, this should not work as I understand?

Reply:
So do these graphs take in numbers as strings? I'm a bit confused :D

Reply:
id doesn't matter for this JS code if its just decimal with ".", or decimal inside qoutes, what matters it has to have dot :D

Reply:
Oh, so I guess that does mean it can't be displayed as currency. Then the way you've set it up here is really the only way to do it :D

Reply:
Well I don't like this part myself, but it works and doesn't seem to increase loading time very much so I left it.